docs/src/main/docs/guides/04_dockerfile.adoc

```diff
@@ -0,0 +1,216 @@
+====
+
+[source,bash]
+.Build the project
+----
+mvn package
+----
+
+[source,bash]
+.Build the Docker image
+----
+docker build -t quickstart-se target
+----
+
+[source,bash]
+.Run the docker container
+----
+docker run --rm -p 8080:8080 quickstart-se:latest
+----
+
+[source,bash]
+.Ping the application
+----
+curl -X GET http://localhost:8080/greet
+----
+
+== Creating a Docker Image with a Custom JRE
+
+This section describes how to build an image with a custom Java 11 JRE using
+ `jlink`.
+
+Replace `src/main/docker/Dockerfile` with the following:
+
+[source,yaml]
+----
+# Multistage Docker build. <1>
+# Stage 1: Build custom Java 11 JRE and put it in /var/tmp/myjre <2>
+FROM openjdk:11-slim AS myjre
+RUN ["jlink", "--compress=2", "--strip-debug", "--no-header-files", \ 
+     "--add-modules", "java.base,java.logging,java.sql,java.desktop,java.management", \
+     "--output", "/var/tmp/myjre"] # <3>
+
+# Work around for https://github.com/docker-library/openjdk/issues/217 <4>
+RUN [ "apt", "update"]
+RUN [ "apt-get", "install", "-y", "binutils"]
+RUN ["strip", "-p", "--strip-unneeded", "/var/tmp/myjre/lib/server/libjvm.so"]
+# End work-around
+
+# Stage 2: Build application image using JRE from Stage 1 <5>
+FROM debian:sid-slim # <6>
+COPY --from=myjre /var/tmp/myjre /opt/jre # <7>
+ENV PATH=$PATH:/opt/jre/bin
+
+RUN mkdir /app
+COPY libs /app/libs
+COPY ${project.artifactId}.jar /app
+
+CMD ["java", "-jar", "/app/${project.artifactId}.jar"]
+----
+<1> This is a multi-stage Docker build. See more info
+ link:{docker-multi-stage-url}[here]
+<2> The first stage that creates our custom JRE.
+<3> The modules listed in this example are for Helidon SE. See below for
+ Helidon MP.
+<4> This is a work-around for {openjdk-docker-217}.
+<5> The final stage that creates the image for our application.
+<6> Use `debian:sid-slim` to match the base image of `openjdk:11-slim`
+<7> Copy the JRE from the image of the first stage `myjre`
+
+[source,bash]
+.Process the new Dockerfile
+----
+mvn process-resources
+----
+
+[source,bash]
+.Build the Docker image
+----
+docker build -t java11-quickstart-se target
+----
+
+[source,bash]
+.Run the docker container
+----
+docker run --rm -p 8080:8080 java11-quickstart-se:latest
+----
+
+[source,bash]
+.Ping the application
+----
+curl -X GET http://localhost:8080/greet
+----
+
+[source,bash]
+.Take a look at the image size
+----
+docker images java11-quickstart-se:latest
+----
+
+[source,bash]
+----
+REPOSITORY             TAG        IMAGE ID         CREATED             SIZE
+java11-quickstart-se   latest     f07a7b8bda78     About a minute ago  136MB
+----
+
+NOTE: ~140MB is less than the pre-built OpenJDK slim JRE images. Results might
+ differ a bit depending on your platform.
+
+=== What about Helidon MP?
+
+For Helidon MP you need to add a couple more modules to the `jlink` command:
+
+[source,yaml,subs="verbatim,attributes"]
+----
+RUN ["jlink", "--compress=2", "--strip-debug", "--no-header-files" \
```

Comment:
Type: There is a comma missing at the end of this line.

Reply:
roger that, will fix.